### PR TITLE
Eng 2080 update list workflow api route to get checks and metrics

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR adds metrics and checks in each item in `list_workflow` route. Additionally, we added some more formalized view structs `OperatorWithResult` and `ArtifactWithResult` which we expect to reuse more in the future for result related application logics.

## Related issue number (if any)
ENG-2080

## Demo:
```
[
    {
        "id": "26ebf1b9-8156-4385-b4aa-51e527d8771b",
        "name": "test_conda",
        "description": "test",
        "created_at": 1670612218,
        "last_run_at": 1670831738,
        "status": "succeeded",
        "engine": "aqueduct",
        "checks": [
            {
                "id": "9654c733-82e1-4b2f-8e81-7b0a61bc5235",
                "name": "greater_than_20",
                "description": "",
                "spec": {
                    "type": "check",
                    "check": {
                        "level": "warning",
                        "function": {
                            "type": "file",
                            "language": "Python",
                            "granularity": "table",
                            "storage_path": "operator-858eb4b5-a7eb-47d1-90d1-bb2f22fb36f7",
                            "custom_args": ""
                        }
                    }
                },
                "inputs": null,
                "outputs": null,
                "result": {
                    "id": "3e2d4e9e-3317-45c6-b2ef-74adc62a3b32",
                    "exec_state": {
                        "user_logs": {
                            "stdout": "",
                            "stderr": ""
                        },
                        "status": "succeeded",
                        "failure_type": null,
                        "error": null,
                        "timestamps": {
                            "registered_at": null,
                            "pending_at": "2022-12-12T07:55:38.218567541Z",
                            "running_at": "2022-12-12T07:55:57.252812395Z",
                            "finished_at": "2022-12-12T07:56:01.623857962Z"
                        }
                    }
                }
            }
        ],
        "metrics": [
            {
                "id": "336ef994-3564-4399-bdfc-b789de4df622",
                "name": "row_count artifact",
                "description": "",
                "type": "numeric",
                "from": "00000000-0000-0000-0000-000000000000",
                "to": null,
                "result": {
                    "id": "28f5a433-8d5b-4bea-ae7e-78150ad8ad56",
                    "serialization_type": "json",
                    "content_path": "6e2ca54f-f27c-4580-910e-c37b63cf04ee",
                    "content_serialized": "100",
                    "exec_state": {
                        "user_logs": {
                            "stdout": "",
                            "stderr": ""
                        },
                        "status": "succeeded",
                        "failure_type": null,
                        "error": null,
                        "timestamps": {
                            "registered_at": null,
                            "pending_at": "2022-12-12T07:55:38.218563226Z",
                            "running_at": "2022-12-12T07:55:52.32882674Z",
                            "finished_at": "2022-12-12T07:55:56.751647274Z"
                        }
                    }
                }
            }
        ]
    }
]
```

## Checklist before requesting a review
We will add unit test in separate PRs
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/cmd/server/handler/list_workflows.go [hunk 1]
```diff
@@ -95,6 +142,35 @@ func (h *ListWorkflowsHandler) Perform(ctx context.Context, interfaceArgs interf
...
+			for _, checkResult := range checkResultsByDAGResultID[latestStatus.ResultID] {
+				response.Checks = append(
+					response.Checks,
+					*operator.NewResultResponseFromDBView(&checkResult),
+				)
+			}
+
+			for _, metricResult := range metricResultsByDAGResultID[latestStatus.ResultID] {
+				var contentPtr *string = nil
+				storageObj := storage.NewStorage(&metricResult.StorageConfig)
```
Comment (cw75) [inline src/golang/cmd/server/handler/list_workflows.go:154]: Seems that this line should go inside `if metric.Type.IsCompact()`.

**Review by cw75 — APPROVED** [review]
A couple nits otherwise LGTM! I think both this route and the artifact versions route have a lot of for loops going on, and we should create tasks to 1) cleanup any info not needed by the UI and 2) see if there's any code refactor we can do to be more efficient and readable.

File: src/golang/lib/repos/artifact_result.go [hunk 2]
```diff
@@ -43,6 +43,15 @@ type artifactResultReader interface {
 
 	// GetByArtifactBatch returns all results whose artifact is among artifactIDs
 	GetByArtifactBatch(ctx context.Context, artifactIDs []uuid.UUID, DB database.Database) ([]models.ArtifactResult, error)
+
+	// GetWithArtifactOfMetricsByDAGResultBatch returns all results together with artifact
+	// for all results generated by metric operator, where the DAGResultID is among the given
+	// list of IDs.
+	GetWithArtifactOfMetricsByDAGResultBatch(
```
[the remark this answers is not in the document]
Comment (likawind) [reply src/golang/lib/repos/artifact_result.go:50]: Let's address this in a separate PR so that we can unblock frontend first

**Comment by likawind** [comment]
@cw75 yeah I think we can actually reuse `OpWithResult` and `ArtfWithResult`, and turn their underlying query in views. This way there's a lot in `get_artifact_versions` we can simplify

**Pull request merged**